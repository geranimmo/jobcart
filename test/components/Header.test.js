import React from 'react';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import expect from 'expect';
// import sinon from 'sinon';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ConnectedHeader, { Header } from '../../src/components/header';

describe('>>> H E A D E R ---- Test & Snapshot <<<', () => {
	const mockStore = configureMockStore([thunk]);
	let wrapper, store;

	const propHeader = {
		onCart: true,
		ShoppingCart: [
			{ id: 'Classic', buy_time: 1537425742947 },
			{ id: 'Premium', buy_time: 1537425742948 }
		],
		Profile: {
			client_id: 1,
			first_name: "Default",
			last_name: "Client",
			business_name: "Default",
			phone_number: "+612529291002",
			client_logo: "client-default-logo.png",
			client_special: []
		},
		UserLogin: true,
		showCart: jest.fn()
	};

	beforeEach(() => {
		store = mockStore(propHeader);
		wrapper = mount(
			<Provider store={store}>
				<Router>
					<ConnectedHeader {...propHeader}/>
				</Router>
			</Provider>
		);
	});

	it('+++ Render the Header component +++', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('+++ Should return of total Cart length in Cart notification +++', () => {
		expect(wrapper.find('div#openCartView').text())
			.toEqual(`${propHeader.ShoppingCart.length}`);
	});

	it('+++ Capturing Snapshot of Header component +++', () => {
		const renderedValue = renderer.create(
			<Provider store={store}>
				<Router>
					<ConnectedHeader {...propHeader}/>
				</Router>
			</Provider>
		).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});

	it('+++ Should call showCart() when click cart button +++', () => {
		const history = { push: jest.fn() };
		const wrappers = shallow(<Header {...propHeader} history={history}/>);
		const showCart = jest.spyOn(wrappers.instance(), 'showCart');

		wrappers.instance().forceUpdate();
		wrappers.update();
		wrappers.find('div#openCartView').first().simulate('click');
		expect(showCart.mock.calls.length).toBe(1);
	});

	it('+++ Should call closeCart() when click close cart button +++', () => {
		const history = { push: jest.fn() };
		const wrappers = shallow(<Header {...propHeader} history={history}/>);
		const showCart = jest.spyOn(wrappers.instance(), 'closeCart');

		wrappers.instance().forceUpdate();
		wrappers.update();
		wrappers.find('div#closeCartView').first().simulate('click');
		expect(showCart.mock.calls.length).toBe(1);
	});
});