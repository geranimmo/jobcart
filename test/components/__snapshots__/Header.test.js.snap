// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> H E A D E R ---- Test & Snapshot <<< +++ Capturing Snapshot of Header component +++ 1`] = `
<header
  className="header__active"
  id="header"
>
  <nav
    className="nav__left"
  >
    <a
      aria-current={null}
      href="/home"
      onClick={[Function]}
    >
      <div
        className="seek__logo"
      />
    </a>
    <div
      className="client__logo__wrapper"
    >
      <img
        alt="Default"
        src="client-default-logo.png"
      />
    </div>
  </nav>
  <nav
    className="nav__right"
  >
    <div
      className="menu__btn__icon hidden"
      id="openCartView"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 3l-.743 2h-1.929l-3.474 12h-13.239l-4.615-11h16.812l-.564 2h-13.24l2.937 7h10.428l3.432-12h4.195zm-15.5 15c-.828 0-1.5.672-1.5 1.5 0 .829.672 1.5 1.5 1.5s1.5-.671 1.5-1.5c0-.828-.672-1.5-1.5-1.5zm6.9-7-1.9 7c-.828 0-1.5.671-1.5 1.5s.672 1.5 1.5 1.5 1.5-.671 1.5-1.5c0-.828-.672-1.5-1.5-1.5z"
        />
      </svg>
      <div
        className="cart__notif"
      >
        2
      </div>
    </div>
    <div
      className="menu__btn__icon "
      id="closeCartView"
      onClick={[Function]}
    >
      <svg
        className="icon__close"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M23.954 21.03l-9.184-9.095 9.092-9.174-2.832-2.807-9.09 9.179-9.176-9.088-2.81 2.81 9.186 9.105-9.095 9.184 2.81 2.81 9.112-9.192 9.18 9.1z"
        />
      </svg>
    </div>
  </nav>
</header>
`;
